src/persist/src/nemesis/generator.rs

```diff
@@ -143,6 +148,17 @@ impl ReqGenerator {
         let ts = stream_sealed_ts + rng.gen_range(0..5);
         Req::Seal(SealReq { stream, ts })
     }
+    
+    fn allow_compaction(rng: &mut SmallRng, state: &mut GeneratorState) -> Req {
+        let stream = rng_stream(rng);
+        let stream_compacted_ts = state
+            .compacted_by_stream
+            .get(&stream)
+            .copied()
+            .unwrap_or_default();
+        let ts = stream_compacted_ts + rng.gen_range(0..5);
```

Comment:
I think this would be more likely to generate interesting traffic if it was based on a +- from the sealed ts for the stream (like write does)

Reply:
So the only valid arguments for `allow_compaction` are within `(compaction_frontier, sealed_frontier)`. The approach here tries to bias for "be in advance of the compaction frontier 80% of time".

I suppose the most neutral policy I can think of is choose a timestamp with equal probability below, within, or above this interval when its non-empty (when sealed_frontier - compaction_frontier > 1) and otherwise with an even probability of picking below or above the valid interval. But I guess that gets tricky when thinking about below the interval because its bounded by 0. I'll think about this some more, but does the policy I described above match your understanding of interesting?

Reply:
did a simpler thing

50% of the time pick a time based off the current since frontier + [0, 5)
50% of the time pick a time based off the current seal frontier + [0, 5)